New resource azurerm_express_route_circuit_connection (#11303)
terraform-providers/terraform-provider-azurerm: website/docs/r/express_route_circuit_connection.html.markdown

  secondary_peer_address_prefix = "123.0.0.4/30"
  vlan_id                       = 300
}

resource "azurerm_express_route_circuit_peering" "example2" {
  peering_type                  = "AzurePrivatePeering"
  express_route_circuit_name    = azurerm_express_route_circuit.example2.name
  resource_group_name           = azurerm_resource_group.example.name
  peer_asn                      = 100
  primary_peer_address_prefix   = "123.0.0.0/30"
  secondary_peer_address_prefix = "123.0.0.4/30"
  vlan_id                       = 300
}

resource "azurerm_express_route_circuit_connection" "example" {
  name                = "example-expressroutecircuitconn"
  peering_id          = azurerm_express_route_circuit_peering.example.id
  peer_peering_id     = azurerm_express_route_circuit_peering.example2.id
  address_prefix_ipv4 = "192.169.8.0/29"
  authorization_key   = "00000000-0000-0000-0000-000000000000"
  address_prefix_ipv6 = "2002:db01::/125"
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Express Route Circuit Connection. Changing this forces a new Express Route Circuit Connection to be created.

> * `peering_id` - (Required) The ID of the Express Route Circuit Private Peering within which this Express Route Circuit Connection should be created. Changing this forces a new Express Route Circuit Connection to be created.

Reviewer: ```suggestion
* `peering_id` - (Required) The ID of the Express Route Circuit Private Peering that this Express Route Circuit Connection connects with. Changing this forces a new Express Route Circuit Connection to be created.
```

Reviewer 2: Updated

  
* `peer_peering_id` - (Required) The ID of the peered Express Route Circuit Private Peering. Changing this forces a new Express Route Circuit Connection to be created.
  

> * `address_prefix_ipv4` - (Required) The IPv4 address space to carve out customer addresses for tunnels.

Reviewer: Can we use the same tone as `address_prefix_ipv6`?

```suggestion
* `address_prefix_ipv4` - (Required) The IPv4 address space from which to allocate customer address for global reach.
```

Reviewer 2: Updated

---

* `authorization_key` - (Optional) The authorization key which is associated with the Express Route Circuit Connection.

> * `address_prefix_ipv6` - (Required) The IPv6 address space to carve out customer addresses for global reach.

Reviewer 2: It should be optional property.
